community/kernel/src/main/java/org/neo4j/kernel/impl/api/state/TxState.java

```diff
@@ -353,18 +348,32 @@ public boolean hasChanges()
     @Override
     public Iterable<NodeState> modifiedNodes()
     {
-        return NODE_STATE.values( this );
+        return nodeStatesMap == null ? Iterables.empty() : Iterables.cast( nodeStatesMap.values() );
     }
 
     private DiffSets<Long> getOrCreateLabelStateNodeDiffSets( int labelId )
     {
-        return LABEL_STATE.getOrCreate( this, labelId ).getOrCreateNodeDiffSets();
+        if ( labelStatesMap == null )
+        {
+            labelStatesMap = Primitive.intObjectMap();
+        }
+        return labelStatesMap.computeIfAbsent( labelId, unused -> new DiffSets<>() );
+    }
+
+    private ReadableDiffSets<Long> getRoLabelStateNodeDiffSets( int labelId )
```

Comment:
what does that 'Ro' part stand's for here and in other methods?

Reply:
It meant to be read-only, but I'll remove it.